cpp/src/datetime/datetime_ops.cu

```diff
@@ -453,90 +440,70 @@ std::unique_ptr<column> extract_year(column_view const& column,
                                      rmm::cuda_stream_view stream,
                                      rmm::device_async_resource_ref mr)
 {
-  return detail::apply_datetime_op<
-    detail::extract_component_operator<detail::datetime_component::YEAR>,
-    cudf::type_id::INT16>(column, stream, mr);
+  return detail::extract_datetime_component(column, datetime_component::YEAR, stream, mr);
 }
 
 std::unique_ptr<column> extract_month(column_view const& column,
```

Comment:
Do we plan to deprecate the specific functions like this in favor of requiring the enum-based call?

Reply:
I believe in the original issue we discussed moving in that direction, not sure how/on what timeline we want that to happen.

Reply:
Let’s open a follow-up PR adding deprecations targeting 24.12.